lost/src/test/java/com/mapzen/android/lost/internal/FusedLocationProviderApiImplTest.java

```diff
@@ -219,7 +219,17 @@ public void setMockMode_shouldNotRegisterDuplicateListeners() throws Exception {
         api.requestLocationUpdates(request, listener);
         api.setMockMode(false);
         api.requestLocationUpdates(request, listener);
-        assertThat(shadowLocationManager.getRequestLocationUpdateListeners()).hasSize(2);
+        assertThat(shadowLocationManager.getRequestLocationUpdateListeners()).hasSize(4);
```

Comment:
It seems like `shadowLocationManager.getRequestLocationUpdateListeners()` should still return `2` if the original listeners were properly removed. There should be one listener for the GPS provider and one for the network provider.

I stepped through this test with the debugger. What seems to be happening is the first time `requestLocationUpdates(...)` is called no listeners are registered b/c mock mode is true. When mock mode is set to `false` then `toggleAllEngines()` registers two listeners (GPS + network) with the `LocationManager`. Then when `requestLocationUpdates(...)` is called for the second time the same two listeners are registered with the `LocationManager` yet again.

I'm not sure what the implications are of registering the same listener twice with the `LocationManager`. Perhaps it doesn't matter. Or perhaps we can set a flag to when the engine has been registered with the `LocationManager` to ensure the same engine does not get registered more than once.